freva-rest/src/databrowser_api/endpoints.py

```diff
@@ -235,3 +261,65 @@ async def load_data(
         status_code=status_code,
         media_type="text/plain",
     )
+
+
+@app.post(
+    "/api/databrowser/userdata",
+    status_code=status.HTTP_202_ACCEPTED,
+    tags=["User Data"],
+)
+async def post_user_data(
+    request: Annotated[AddUserDataRequestBody, Body(...)],
+    current_user: TokenPayload = Depends(auth.required),
+) -> Dict[str, str]:
+    """Add user metadata into the Apache Solr search system and MongoDB."""
+
+    solr_instance = Solr(server_config)
+    try:
+        validated_user_metadata = await solr_instance._validate_user_metadata(
+            request.user_metadata
+        )
+        status_msg = await solr_instance.add_user_metadata(
+            current_user.preferred_username,  # type: ignore
+            validated_user_metadata,
+            facets=request.facets
+        )
+    except Exception as e:
```

Comment:
Could you distinguish at least for validation errors? That way we get a 422 on failed validations. The rest can be 500